services/src/main/java/org/keycloak/protocol/saml/SamlProtocol.java

```diff
@@ -748,6 +759,38 @@ protected Response buildLogoutResponse(UserSessionModel userSession, String logo
     public Response backchannelLogout(UserSessionModel userSession, AuthenticatedClientSessionModel clientSession) {
         ClientModel client = clientSession.getClient();
         SamlClient samlClient = new SamlClient(client);
+
+        // real backchannel logout if SOAP binding is supported (#9548)
+        String soapLogoutUrl = getLogoutServiceUrl(session, client, SAML_SOAP_BINDING, true);
+        if (soapLogoutUrl != null) {
+            try {
+                LogoutRequestType logoutRequest = createLogoutRequest(soapLogoutUrl, clientSession, client);
+                Document samlLogoutRequest = createBindingBuilder(samlClient, false).postBinding(SAML2Request.convert(logoutRequest)).getDocument();
+                SOAPMessage soapResponse = Soap.createMessage().addToBody(samlLogoutRequest).call(soapLogoutUrl);
+                Document logoutResponse = Soap.extractSoapMessage(soapResponse);
+                SAMLDocumentHolder samlDoc = SAML2Response.getSAML2ObjectFromDocument(logoutResponse);
+                if (!(samlDoc.getSamlObject() instanceof StatusResponseType)) {
+                    throw new RuntimeException("Logout response received from client is not an StatusResponseType");
+                }
+                StatusResponseType logoutStatus = (StatusResponseType) samlDoc.getSamlObject();
+                if (!Objects.equals(logoutStatus.getStatus().getStatusCode().getValue().toString(), JBossSAMLURIConstants.STATUS_SUCCESS.get())) {
```

Comment:
@basbeu About NPE: if there is an NPE, then the client has provided a response that is not valid according to the SAML protocol. I've slightly adapted the equality check to be a bit safer, and to do the same as an existing piece of code (https://github.com/keycloak/keycloak/blob/168734b817da2138548bacba973204e80f566b2c/services/src/main/java/org/keycloak/broker/saml/SAMLEndpoint.java#L616)